src/sddp.jl

```diff
@@ -58,16 +86,86 @@ struct SDDP <: SOI.AbstractAlgorithm
     options::Dict
 end
 
-# TODO!
-struct CutGenerator end
+################################################################################
+# Cuts generator
+################################################################################
 
+abstract type AbstractCutGenerator end
 
-# TODO
-function gencuts end
+####################
+struct AverageCutGenerator <: AbstractCutGenerator end
+
+# TODO: state `x` is missing
+
+function gencuts(::AverageCutGenerator, sp, node, pos, pool)
+    avgλ = zeros()
+    avgβ = 0.
+
+    ξ = probalaw(node)
+    wghts = Scenarios.weights(ξ)
```

Comment:
This is assuming noise and transition independent, but it is not leveraging this fact to make computation faster. I suggest having probabilities on transition directly.